server/models/db/timeslot.js

```diff
@@ -0,0 +1,24 @@
+const _ = require('lodash');
+
+const { db } = require('../../lib/connection');
+const Tables = require('../../constants/tables.json');
+//const TimeslotStatus = require('../../constants/timeslot-status.json');
+
+// This method returns the timeslot records stored in the timeslot table 
+//Note the parameters, as this method, if given specific start time and specific end time to return, wil only return timeslots within the start and end time parameters 
+//If one or both ofthe parameters is null or not vlaid, the method will return all timeslots in the timeslot table  
+async function getPossibleTimeslots(specificStartTime, specificEndTime) {
+    if (_.isNil(specificStartTime) || _.isNil(specificEndTime)) {
```

Comment:
Hi Peter,
When checking the `specificStartTime` and `specificEndTime`, the moment library can also be leveraged to ensure that the parameters passed in are valid date strings rather than just checking null. For instance, if this method was called with `hello` for some reason, then currently, the query would still execute and have a condition on the `start_time >= 'hello'` which may not be desired.

Reply:
This comment is relevant to a part of the development no longer part of this PR.